packages/engine/Source/Core/JulianDate.js

```diff
@@ -783,40 +783,16 @@ JulianDate.toIso8601 = function (julianDate, precision) {
 
   let millisecondStr;
 
-  if (!defined(precision) && millisecond !== 0) {
-    //Forces milliseconds into a number with at least 3 digits to whatever the default toString() precision is.
-    millisecondStr = (millisecond * 0.01).toString().replace(".", "");
-    return `${year.toString().padStart(4, "0")}-${month
-      .toString()
-      .padStart(2, "0")}-${day
-      .toString()
-      .padStart(2, "0")}T${hour
-      .toString()
-      .padStart(2, "0")}:${minute
-      .toString()
-      .padStart(2, "0")}:${second
-      .toString()
-      .padStart(2, "0")}.${millisecondStr}Z`;
-  }
-
-  //Precision is either 0 or milliseconds is 0 with undefined precision, in either case, leave off milliseconds entirely
-  if (!defined(precision) || precision === 0) {
-    return `${year.toString().padStart(4, "0")}-${month
-      .toString()
-      .padStart(2, "0")}-${day
-      .toString()
-      .padStart(2, "0")}T${hour
-      .toString()
-      .padStart(2, "0")}:${minute
-      .toString()
-      .padStart(2, "0")}:${second.toString().padStart(2, "0")}Z`;
-  }
-
-  //Forces milliseconds into a number with at least 3 digits to whatever the specified precision is.
-  millisecondStr = (millisecond * 0.01)
-    .toFixed(precision)
-    .replace(".", "")
-    .slice(0, precision);
+  if (millisecond !== 0 || defined(precision)) {
+    if (!defined(precision)) {
+      precision = 7; // Default precision if not provided
+    }
+    // Forces milliseconds into a number with the specified precision, without scientific notation
+    const fractionalSeconds = (second + millisecond * 0.001).toFixed(precision);
+    const parts = fractionalSeconds.split('.');
+    millisecondStr = parts.length > 1 ? parts[1] : '';
```

Comment:
Can you expand on why you're adding the seconds here then splitting on the `.`? I think there may potentially be other, better, ways to get the decimal part of a number if that's the desire.

Also the original issue called out that `toFixed` does not necessarily eliminate the issue of getting scientific notation when converting a number to string if the precision is right (see [the docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toPrecision)). Does this addition avoid that somehow? or is that not accounted for?

Reply:
I am splitting on . to separate the fractional part (milliseconds) from the whole seconds.
I am not getting scientific notations for very small numbers.

Reply:
My initial thought is that splitting on `.` is a bad way to manipulate numbers but looking into it a bit this seems like a fine way to do it. However, I'm still unclear about why you're adding the seconds to it first if you just strip them off.
It's possible there's a more simple solution that _does_ include combining the seconds and milliseconds that looks something like this `(n * 0.001 + 45).toFixed(precision).padStart(2 + precision + 1, '0')`. Just pay special attention to the padding for the seconds.

Also it looks like we have constants for some of these "magic numbers", please swap out `0.001` with `TimeConstants.SECONDS_PER_MILLISECOND` as we use elsewhere in this code.